core/node/proof_data_handler/src/request_processor.rs

```diff
@@ -61,11 +64,32 @@ impl RequestProcessor {
             None => return Ok(Json(ProofGenerationDataResponse::Success(None))), // no batches pending to be proven
         };
 
-        let blob = self
+        let mut vm_run_data: VMRunWitnessInputData = self
             .blob_store
             .get(l1_batch_number)
             .await
             .map_err(RequestProcessorError::ObjectStore)?;
+        let merkle_paths: WitnessInputMerklePaths = self
+            .blob_store
+            .get(l1_batch_number)
+            .await
+            .map_err(RequestProcessorError::ObjectStore)?;
+
+        let previous_batch_metadata = self
+            .pool
+            .connection()
+            .await
+            .unwrap()
+            .blocks_dal()
+            .get_l1_batch_metadata(L1BatchNumber(l1_batch_number.checked_sub(1).unwrap()))
+            .await
+            .unwrap()
+            .expect("No metadata for previous batch");
+
+        vm_run_data.previous_root_hash = Some(previous_batch_metadata.metadata.root_hash);
+        vm_run_data.previous_meta_hash =
+            Some(previous_batch_metadata.metadata.meta_parameters_hash);
+        vm_run_data.previous_aux_hash = Some(previous_batch_metadata.metadata.aux_data_hash);
```

Comment:
This doesn't look right. I would move these hashes from vm run data to a separate struct